acceptance/launcher_test.go

```diff
@@ -0,0 +1,215 @@
+package acceptance
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+var (
+	launchDockerContext = filepath.Join("testdata", "launcher")
+	launcherBinaryDir   = filepath.Join("acceptance", "testdata", "launcher", "container", "cnb", "lifecycle")
+	launchImage         = "lifecycle/acceptance/launcher"
+)
+
+func TestLauncher(t *testing.T) {
+	buildLauncher(t)
+	buildLaunchImage(t)
+	defer removeLaunchImage(t)
+	spec.Run(t, "acceptance", testLauncher, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testLauncher(t *testing.T, when spec.G, it spec.S) {
+	when("no args and no default", func() {
+		it("launches web", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage)
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "Executing web process-type"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute web:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+
+	when("no args and CNB_PROCESS_TYPE", func() {
```

Comment:
It doesn't look like we have any tests around setting `CNB_LAYERS_DIR` and `CNB_APP_DIR`. I suppose that would require either a separate container fixture, or some manipulation with `docker` before running the tests.

Reply:
Yeah, it would require either a different fixture, or multiple dirs we can switch between in the same fixture. I'll add a test.